_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,355 @@
+
+Login to your DigitalOcean account, and in the left menu choose **Kubernetes**. Then choose your cluster, and on the cluster page, choose **Resources**. In the _LOAD BALANCERS_ list, choose the single load balancer that must have been created. On the load balancer page, choose the **Settings** tab. Click **Edit** in the _Forwarding Rules_. Add a forwarding rule for HTTPS in port 443, and in the certificate drop-down, choose **New certificate**.
+
+{% img blog/jhipster-digital-ocean/do-new-certificate.png alt:"DigitalOcean new certificate form" width:"500" %}{: .center-image }
+
+In the **New Certificate** form, choose the **Use Let's Encrypt** tab, and then in the domain box, choose **Add new domain**. Then enter your domain name, and list other subdomains to include. Add a name for the certificate and click **Generate Certificate**.
+
+{% img blog/jhipster-digital-ocean/do-new-domain.png alt:"DigitalOcean new domain form" width:"500" %}{: .center-image }
+
+Back in the _Forwarding rules_ page, check the generated certificate is selected, and forward the traffic to the droplet HTTP port where the gateway is running. Tick the checkbox **Create DNS records for all the new Let's Encrypt certificates** and then **Save** the forwarding settings.
+
+{% img blog/jhipster-digital-ocean/do-forwarding.png alt:"DigitalOcean load balancer forwarding settings" width:"800" %}{: .center-image }
+
+Finally, in the SSL section of the settings, tick the checkbox **Redirect HTTP to HTTPS**.
+
+{% img blog/jhipster-digital-ocean/do-ssl-redirect.png alt:"DigitalOcean load balancer ssl redirect settings" width:"800" %}{: .center-image }
+
+Once again, update the redirect URIs in Okta to allow the newly configured domain. Set up the URIs with `https://yourDomain` prefix.
+
+Test the configuration by navigating to `http://yourDomain`. First, the load balancer should redirect to HTTPs, and then the gateway should redirect to the Okta sign-in page.
+
+## Learn more about JHipster and cloud deployment
+
+And this was a brief walk-through for deploying JHipster to a DigitalOcean's managed Kubernetes cluster. Some important topics for production deployment were not covered in this post, to focus on DigitalOcean resource requirements in particular. Among them, external configuration storage with Spring Cloud Config and Git was not covered, secrets encryption both for the cloud configuration and Kubernetes configuration also was not covered. You can learn about these good practices in the first post of this cloud deployment series: [Kubernetes to the Cloud with Spring Boot and JHipster](/blog/2021/06/01/kubernetes-spring-boot-jhipster). Keep learning, and for more content on JHipster, check out the following links:
+
+- [Introducing Spring Native for JHipster: Serverless Full-Stack Made Easy](/blog/2022/03/03/spring-native-jhipster)
```

Comment:
We should add Andrew's post to this list. https://developer.okta.com/blog/2022/05/05/kubernetes-microservices-azure

Reply:
Added